driver-reactive-streams/src/main/com/mongodb/reactivestreams/client/AggregationBatchCursorPublisher.java

```diff
@@ -0,0 +1,28 @@
+/*
+ * Copyright 2008-present MongoDB, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.mongodb.reactivestreams.client;
+
+/**
+ * Extends the async batch cursor interface to include information included in an aggregate or getMore response.
+ *
+ * @param <T> The type of documents the cursor contains
+ * @mongodb.driver.manual ../meta-driver/latest/legacy/mongodb-wire-protocol/#wire-op-get-more OP_GET_MORE
+ * @since 4.2
+ */
+public interface AggregationBatchCursorPublisher<T> extends BatchCursorPublisherGeneric<AggregationBatchCursor<T>, T> {
```

Comment:
Type juggling - so we can have the BatchCursor type and the result type defined.